swift/Samples/Tutorial/Tutorial2.md

```diff
@@ -0,0 +1,597 @@
+
+        super.init(nibName: nil, bundle: nil)
+    }
+    // ... the rest of the implementation ...
+```
+
+Then, update the `RootWorkflow` to return a `BackStackScreen` with a list of back stack items:
+
+```swift
+// Don't forget to import `BackStackContainer` to be able to use `BackStackScreen`.
+import BackStackContainer
+
+// ...snipped...
+
+// MARK: Rendering
+
+extension RootWorkflow {
+
+    typealias Rendering = BackStackScreen
+
+    func render(state: RootWorkflow.State, context: RenderContext<RootWorkflow>) -> Rendering {
+        // Create a sink to handle the back action from the TodoListWorkflow to logout.
+        let sink = context.makeSink(of: Action.self)
+
+        // Our list of back stack items. Will always include the "WelcomeScreen".
+        var backStackItems: [BackStackScreen.Item] = []
+
+        let welcomeScreen = WelcomeWorkflow()
+            .mapOutput({ output -> Action in
+                switch output {
+                // When `WelcomeWorkflow` emits `didLogin`, turn it into our `login` action.
+                case .didLogin(name: let name):
+                    return .login(name: name)
+                }
+            })
+            .rendered(with: context)
+
+        let welcomeBackStackItem = BackStackScreen.Item(
+            key: "welcome",
+            screen: welcomeScreen,
+            // Hide the navigation bar.
+            barVisibility: .hidden)
+
+        // Always add the welcome back stack item.
+        backStackItems.append(welcomeBackStackItem)
+
+        switch state {
+        // When the state is `.welcome`, defer to the WelcomeWorkflow.
+        case .welcome:
+            // We always add the welcome screen to the backstack, so this is a no op.
+            break
+
+        // When the state is `.todo`, defer to the TodoListWorkflow.
+        case .todo(name: let name):
+
+            let todoListScreen = TodoListWorkflow()
+                .rendered(with: context)
+
+            let todoListBackStackItem = BackStackScreen.Item(
+                key: "todoList",
+                screen: todoListScreen,
+                // Specify the title, back button, and right button.
+                barContent: BackStackScreen.BarContent(
+                    title: "Welcome \(name)",
+                    // When `back` is pressed, emit the .logout action to return to the welcome screen.
+                    leftItem: .button(.back(handler: {
+                        sink.send(.logout)
+                    })),
+                    rightItem: .none))
+
+            // Add the TodoListScreen to our BackStackItems.
+            backStackItems.append(todoListBackStackItem)
+        }
+
+        // Finally, return the BackStackScreen with a list of BackStackScreen.Items
+        return BackStackScreen(items: backStackItems)
+    }
+}
+```
+
+![Welcome to Todo List](images/welcome-to-todolist.gif)
+
+Neat! We can now login and logout, and show the name entered as our title!
```

Comment:
The verb forms are separate words, _We can log in and log out,…_

Didn't you teach me that?